renderer/src/helpers/hooks/useEditingHistoryManager.ts

```diff
@@ -56,34 +56,32 @@ const undoReducer = <T>(
   }
 };
 
-const useUndoManager = (initialValue: string) => {
-  const [state, dispatch] = useReducer(undoReducer, {
+const useEditingHistoryManager = <T>(initialValue: T) => {
+  const undoState: EditingHistoryState<T> = {
     undoStack: [initialValue],
     redoStack: [],
-  });
-
-  const setValue = useCallback(
-    (value) => {
-      dispatch({ type: "SET_VALUE", value });
-    },
-    [dispatch]
-  );
+  };
+  const [state, dispatch] = useReducer(EditingHistoryReducer<T>, undoState);
 
-  const undo = useCallback(() => {
-    dispatch({ type: "UNDO" });
+  const doSetValue = (value) => dispatch({ type: "SET_VALUE", value });
```

Comment:
Could you please create "actions" folder inside the src folder and put a file "editingActions" and create actions?
Because { type: "SET_VALUE", value } this should be as action.

Reply:
Resolved